virtual_rainforest/models/soil/soil_model.py

```diff
@@ -132,29 +147,122 @@ def spinup(self) -> None:
+
+    def integrate_soil_model(self) -> Dataset:
+        """Function to integrate the soil model.
+
+        For now a single integration will be used to advance the entire soil module.
+        However, this might get split into several separate integrations in future (if
+        that is feasible).
+
+        This function unpacks the variables that are to be integrated into a single
+        numpy array suitable for integration. This order of this numpy array is as
+        follows: [low_molecular_weight_c, mineral_associated_om].
+
+        Returns:
+            A data array containing the new pool values (i.e. the values at the final
+            time point)
+
+        Raises:
+            IntegrationError: When the integration cannot be successfully completed.
+        """
+
+        # Find number of grid cells integration is being performed over
+        no_cells = self.data.grid.n_cells
+
+        # Extract update interval
+        update_time = self.update_interval.astype("timedelta64[s]").astype(float)
+        # Convert from second to day units
+        t_span = (0.0, update_time / (60.0 * 60.0 * 24.0))
+
+        # Construct vector of initial values y0
+        y0 = np.concatenate(
+            [
+                self.data["low_molecular_weight_c"].to_numpy(),
+                self.data["mineral_associated_om"].to_numpy(),
+            ]
+        )
+
+        # Carry out simulation
+        output = solve_ivp(
+            construct_full_soil_model,
+            t_span,
+            y0,
+            args=(
+                self.data,
+                no_cells,
+            ),
+        )
+
+        # Check whether or not integration succeeded
+        if output.success:
+            return Dataset(
+                data_vars=dict(
+                    new_lmwc=DataArray(output.y[:no_cells, -1], dims="cell_id"),
+                    new_maom=DataArray(output.y[no_cells:, -1], dims="cell_id"),
+                )
+            )
+        else:
+            LOGGER.error(
+                "Integration of soil module failed with following message: %s"
+                % str(output.message)
+            )
+            raise IntegrationError()
+
+
+def construct_full_soil_model(
+    t: float, pools: np.ndarray, data: Data, no_cells: int
+) -> np.ndarray:
+    """Function that constructs the full soil model in a solve_ivp friendly form.
+
+    Args:
+        pools: And area containing all soil pools in a single vector
+        data: The data object, used to populate the arguments i.e. pH and bulk density
+        no_cells: Number of grid cells the integration is being performed over
+
+    Returns:
+        The rate of change for each soil pool
+    """
+
+    pool_rate_of_changes = calculate_soil_carbon_updates(
+        pools[:no_cells],
+        pools[no_cells:],
+        data["pH"],
+        data["bulk_density"],
+        data["soil_moisture"],
+        data["soil_temperature"],
+        data["percent_clay"],
```

Comment:
In addition to @dalonsoa's change, I've a vague preferences for using named keywords, particularly with long arguments? Here, the data keys makes it pretty obvious, but being explicit about what the different inputs are seems more defensive?